theme/ckeditor5-ui/components/dropdown/listdropdown.css

```diff
@@ -5,28 +5,59 @@
 
 @import "../../../mixins/_rounded.css";
 
-.ck.ck-dropdown .ck-dropdown__panel .ck-list {
-	/* Disabled radius of top-left border to be consistent with .dropdown__button
-	https://github.com/ckeditor/ckeditor5/issues/816 */
-	@mixin ck-rounded-corners {
-		border-top-left-radius: 0;
+.ck.ck-dropdown .ck-dropdown__panel .ck-list .ck-list__item .ck-button {
+	border-radius: 0;
+}
+
+.ck.ck-dropdown .ck-dropdown__panel {
+	&.ck-dropdown__panel_sw > .ck-list .ck-list__item:first-child .ck-button {
+		@mixin ck-rounded-corners {
+			border-top-right-radius: 0;
+			border-bottom-right-radius: 0;
+			border-bottom-left-radius: 0;
```

Comment:
I don't like it... Using this mixin here, repeatedly, makes the code too long and hard to read IMHO. 
I think we should create a similar mixin but for reseting radiuses. Then we can mix it with `ck-rounded-corners` but also use it here to reset those and apply radius only for the one corner. WDYT?

Reply:
This is what the mixin looks like https://github.com/ckeditor/ckeditor5-theme-lark/blob/0115e9440ce152135164ba4e4b6ad831b56ab6fd/theme/mixins/_rounded.css#L11-L20

All it does it `border-radius: var(--ck-border-radius);` and then everything from the brackets. It works as an exclusion.

A mixin for resetting radiuses would... look exactly the same as `@mixin ck-rounded-corners` I guess. So I'm not sure we would gain that much code. [postcss-mixins](https://github.com/postcss/postcss-mixins) does not support conditionals so there's no way to specify which radiuses should be reset.

Reply:
Yeah, I know how the mixin looks like, checked it before ;-) So basically:

```
@mixin ck-rounded-corners {
	border-top-right-radius: 0;
	border-bottom-right-radius: 0;
	border-bottom-left-radius: 0;
}
```

does the same thing as:

```
border-radius: 0;
border-top-left-radius: var(--ck-border-radius);
```

right?

The latter is much simpler to me, so my suggestion is to apply the code above or introduce:

```
@mixin ck-reset-corners {
    border-top-left-radius: var(--ck-border-radius);
}
```

Reply:
It makes sense but honestly, I'm confused by the naming. `ck-rounded-corners` enables rounded corners then you can opt-out from some of them. `ck-reset-corners` also adds rounded corners but it does not make sense standalone without `@mixin-content`.

Maybe we should investigate and see if there's a chance for us to [write mixins in JS](https://github.com/postcss/postcss-mixins#function-mixin). We could have a better rounded-corners mixin then:

```css
/* Enable all */
@mixin ck-rounded-corners;

/* Enable bottom-right and top-left only */
@mixin ck-rounded-corners bottom-right top-left;

/* Do something custom (it only adds selectors) */
@mixin ck-rounded-corners {
    border-top-left-radius: 2cm;
}
```

I'm not sure how this could work with our ultra-modular framework. That's probably a follow-up.

Reply:
I get your point here. So to wrap it all up, you still prefer this:

```
@mixin ck-rounded-corners {
	border-top-right-radius: 0;
	border-bottom-right-radius: 0;
	border-bottom-left-radius: 0;
}
```
over this:

```
border-radius: 0;
border-top-left-radius: var(--ck-border-radius);
```
?